File: backend/app/schemas/me.py
from typing import Any, List, Optional
from pydantic import BaseModel


class MeOverviewOut(BaseModel):
    profile: dict
    stats: dict
    recent_orders: List[dict]
    recent_tickets: List[dict]
    badges: List[dict]


class MeHistoryOut(BaseModel):
    orders: List[dict]
    tickets: List[dict]

File: frontend/src/lib/api/me.ts
'use client';

import { apiClient } from './client';

export function getMeOverview() {
  return apiClient.get<{
    profile: Record<string, unknown>;
    stats: Record<string, number>;
    recent_orders: Record<string, unknown>[];
    recent_tickets: Record<string, unknown>[];
    badges: Record<string, unknown>[];
  }>('/me/overview');
}

export function getMeHistory() {
  return apiClient.get<{ orders: Record<string, unknown>[]; tickets: Record<string, unknown>[] }>('/me/history');
}
